File: src/java/CQueue.java
package java;

import java.util.Stack;

class CQueue {

    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public CQueue() {
        stack1 = new Stack<>();
        stack2 = new Stack<>();


    }

    public void appendTail(int value) {
        stack1.push(value);

    }

    public int deleteHead() {
        if (stack1.isEmpty() && stack2.isEmpty()) return -1;
        if (stack2.isEmpty()) {
            while (!stack1.isEmpty()) {
                stack2.push(stack1.pop());
            }
        }
        return stack2.pop();
    }
}

File: src/design/CQueue.kt
package design

class CQueue() {

    var stack1 = ArrayDeque<Int>()
    var stack2 = ArrayDeque<Int>()

    fun appendTail(value: Int) {
        stack1.addLast(value)
    }

    fun deleteHead(): Int {
        if (stack1.isEmpty() && stack2.isEmpty()) return -1
        if (stack2.isEmpty()) {
            while (!stack1.isEmpty()) {
                stack2.addLast(stack1.removeLast())
            }
            return stack2.removeLast()
        }
        return stack2.removeLast()
    }

}